### ISSUE TITLE
Have the auto update shared progress setting be default

### ISSUE THREAD

**Issue by MeemawHustlin** [MichaelEbert/OblivionProgressTracker#286]
As far as I could tell, this wasn't the default, but I couldn't figure out how to change the code that handles default settings to get it to work. I assume the user wants to generate a link and that's it, so the setting that auto updates their progress to the share link should always be on.

**Comment by MichaelEbert** [comment]
add a `changed |= initProperty(settings, "autoUploadCheck", true);` line. 
If you want to reset everyone's existing setting as well, bump the settings version.

**Comment by MichaelEbert** [comment]
OH the reason why auto upload isn't checked by default iirc is GDPR

**Comment by MichaelEbert** [comment]
don't want to upload it before they click teh upload button

**Comment by MeemawHustlin** [comment]
If that box is checked prior to clicking the Create Share URL button, does it do anything? If not, we could lump all of the consent to sharing data into an alert box after clicking the share url button, including the disclaimer about sharing IP and whatnot. Would that conform to the requirements for GDPR?

**Comment by MichaelEbert** [comment]
yes, if the box is clicked, it will upload progress even before the manual upload button has been clicked.

**Comment by MeemawHustlin** [comment]
Ah okay yeah then I guess what we could do is leave the box unchecked by default, put up an alert when you click the share URL button, and have it say "Confirming Yes will select the Auto Share Progress button and continually upload your progress to the server. You can disable this at any time by unchecking this box in the Advanced Sharing section in the Settings." or something to that effect, then have the confirmation change that setting to be on. That way it's all very safe and clear to people how it works without them having to go to two different locations to change that button.

Maybe down the road some sort of "delete my progress from the share server" type feature would be useful.

**Comment by MichaelEbert** [comment]
does it have to be for v4? cause for v5 we'll just make it so that you have to click the button first.

**Comment by MeemawHustlin** [comment]
It's not completely necessary, I just think having the user only have a single place they go to click on one thing is way cleaner and intuitive than having them go into the advanced options and check the box after already generating a share link. I might be misunderstanding how it works though.

**Comment by MichaelEbert** [comment]
nope you got it. 
I'll see if I can add it in if we have time, but don't want it to hold up the rest of the process.
